src/Chapter2.hs

```diff
@@ -858,8 +893,10 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
-
+--rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind xs = foldl (\x y -> y:x) [] xs  
```

Comment:
Ohh, this is an awesome solution!

You can skip first pattern matching on `[]` as `fold` is covering this case already!